src/synology_dsm/api/core/external_usb.py

```diff
@@ -0,0 +1,245 @@
+        return_data = SynoFormatHelper.megabytes_to_bytes(return_data)
+        if human_readable:
+            return SynoFormatHelper.bytes_to_readable(return_data)
+        return return_data
+
+    @property
+    def device_status(self):
+        """The status of the external USB storage device."""
+        return self._data["status"]
+
+    @property
+    def device_formatable(self):
+        """Whether the external USB storage device can be formatted."""
+        return self._data["formatable"]
+
+    @property
+    def device_progress(self):
+        """The progress the external USB storage device."""
+        return self._data["progress"]
+
+    @property
+    def device_product_name(self):
+        """The product name of the external USB storage device."""
+        return self._data["product"]
+
+    @property
+    def device_manufacturer(self):
+        """The producer name of the external USB storage device."""
+        return self._data["producer"]
+
+    # Partition
+    @property
+    def device_partitions(self) -> list[SynoUSBStoragePartition]:
+        """Returns all partitions of the external USB storage device."""
+        partitions = self._data["partitions"]
+        device_partitions: list[SynoUSBStoragePartition] = []
+        for partition in partitions:
+            device_partitions.append(SynoUSBStoragePartition(partition))
+        return device_partitions
+
+    @property
+    def device_partition_ids(self):
+        """Returns partition ids of the external USB storage device."""
+        partitions = self._data["partitions", []]
+        partition_ids = []
+        for partition in partitions:
+            partition_ids.append(partition["name_id"])
+        return partition_ids
+
+    def get_device_partition(self, partition_id) -> SynoUSBStoragePartition:
+        """Returns a partition of the external USB storage device."""
+        return self._data[partition_id]
+
+    def partitions_all_size_total(self, human_readable=False):
+        """Total size of all parititions of the external USB storage device."""
+        partitions = self.device_partitions
+        if not partitions:
+            return None
+
+        size_total = 0
+        for partition in partitions:
+            partition_size = partition.partition_size_total(False)
+            # Partitions may be reported without a size
+            if partition_size:
+                size_total += partition.partition_size_total(False)
+
+        if human_readable:
+            return SynoFormatHelper.bytes_to_readable(size_total)
+        return size_total
+
+    def partitions_all_size_used(self, human_readable=False):
+        """Total size used of all partitions of the external USB storage device."""
+        partitions = self.device_partitions
+        if not partitions:
+            return None
+
+        size_used = 0
+        for partition in partitions:
+            partition_used = partition.partition_size_used(False)
```

Comment:
no need to set `False` because it is the default

Reply:
Should this
```python
for partition in partitions:
```

be this instead?
```python
for partition in partitions.values():
```

When I tried the first way, partition was just a string like 'usb1p1'.  Using the second way, I got the actual partition object.

Reply:
correct, becasue `partitions` is now a dictionary with `str` as keys and `SynoUSBStoragePartition` as values (_equals the type annotation `dict[str, SynoUSBStoragePartition]`_)